### PR TITLE
MOBILE-1046: Update iOS to SDK 13

### PR DESCRIPTION
**Pull request by BrianBatchelder**
### What do these changes do?
This PR updates the iOS bindings to support SDK 13.

### Why are these changes necessary?
Version 13 modularized the SDK, which requires corresponding changes to the Xamarin iOS bindings.

### How did you verify these changes?
Tested channel registration, push, message center and notification service extension on iOS 13 device. Tested build scripts.

### Anything else a reviewer should know?
- The work also required some minor changes to the iOS SDK, mostly added doc comments for elements that are internal but weren't marked that way. These changes should not require an iOS re-release. They will follow in a separate PR.  
- The work also required enhancements to Xamboni. That will also follow in a separate PR.

### CODE REVIEW

File: UrbanAirship.iOS.Core.nuspec [hunk 1]
```diff
@@ -1,16 +1,16 @@
 <?xml version="1.0" encoding="utf-8"?>
```
Comment (rlepinski) [inline UrbanAirship.iOS.Core.nuspec]: I think we still want a UrbanAirship.iOS.nuspec that depends on Core, Automation, MessageCenter, and ExtendedActions

File: src/AirshipBindings.iOS.Automation/build.gradle [hunk 2]
```diff
@@ -28,7 +28,7 @@ task generateBindings {
```
Comment (rlepinski) [inline src/AirshipBindings.iOS.Automation/build.gradle]: Is this task still valid? we generate bindings with Xamboni now

Comment (BrianBatchelder) [reply src/AirshipBindings.iOS.Automation/build.gradle]: It is not valid. I could remove.

File: UrbanAirship.iOS.Core.nuspec (hunk 1 above)
Comment (BrianBatchelder) [reply UrbanAirship.iOS.Core.nuspec]: OK.

File: src/AirshipBindings.Android.ADM/build.gradle [hunk 3]
```diff
@@ -27,7 +27,7 @@ task syncVersion  {
         def path = new File("$projectDir/Jars/$filename")
         if ( !path.exists() ) {
            // Download aar
-            def url = new URL("http://central.maven.org/maven2/com/urbanairship/android/urbanairship-adm/${airshipProperties.androidVersion}/$filename")
+            def url = new URL("https://repo1.maven.org/maven2/com/urbanairship/android/urbanairship-adm/${airshipProperties.androidVersion}/$filename")
```
Comment (rlepinski) [inline src/AirshipBindings.Android.ADM/build.gradle]: you should use the jcenter repos

File: src/AirshipBindings.iOS.Core/ApiDefinitions.cs [hunk 4]
```diff
@@ -0,0 +1,2852 @@
...
+        [Export("consumedTemplateWithValue:")]
+        UAMediaEventTemplate ConsumedTemplate ([NullAllowed] NSNumber eventValue);
+
+        // - (nonnull UACustomEvent *)createEvent;
+        [Export("createEvent")]
+        UACustomEvent CreateEvent ();
+    }
+
+    // @interface UAModifyTagsAction : UAAction
+    [BaseType(typeof(UAAc
```
Comment (marc-scig) [inline src/AirshipBindings.iOS.Core/ApiDefinitions.cs]: Can you do a search/replace for "Deprecated ?" -> "Deprecated -"

There's a ticket to fix the issue that's currently making the question mark appear.

File: src/AirshipBindings.Android.ADM/build.gradle (hunk 3 above)
Comment (marc-scig) [inline src/AirshipBindings.Android.ADM/build.gradle]: What was the reason for this change? Ryan thinks we should probably use jcenter over maven, though that's probably something we can do later.

File: src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Android/AirshipBindings.NETStandard.Android.csproj [hunk 5]
```diff
@@ -14,7 +14,6 @@
     <AndroidResgenClass>Resource</AndroidResgenClass>
     <MonoAndroidResourcePrefix>Resources</MonoAndroidResourcePrefix>
     <MonoAndroidAssetsPrefix>Assets</MonoAndroidAssetsPrefix>
-    <AndroidUseLatestPlatformSdk>true</AndroidUseLatestPlatformSdk>
```
[the remark this answers is not in the document]
Comment (BrianBatchelder) [reply src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Android/AirshipBindings.NETStandard.Android.csproj]: It looks like it has been [deprecated](https://forums.xamarin.com/discussion/158550/androiduselatestplatformsdk-not-recognised-by-vs-mac-2019). Must have been an automatic Visual Studio change. Same change was made to `AirshipBindings.Portable.Android.csproj`

File: src/AirshipBindings.iOS.Core/ApiDefinitions.cs (hunk 4 above)
Comment (BrianBatchelder) [reply src/AirshipBindings.iOS.Core/ApiDefinitions.cs]: Good catch. Forgot about that one.

File: src/AirshipBindings.Android.ADM/build.gradle (hunk 3 above)
Comment (BrianBatchelder) [reply src/AirshipBindings.Android.ADM/build.gradle]: Maven Central no longer allows `http:`. This is the recommended URL. But Ryan wanted bintray, so I've changed it to bintray.

Comment (BrianBatchelder) [reply src/AirshipBindings.Android.ADM/build.gradle]: Maven Central no longer allows `http:`. This is the recommended URL. But Ryan wanted bintray, so I've changed it to bintray.

File: src/AirshipBindings.iOS.Core/ApiDefinitions.cs (hunk 4 above)
Comment (BrianBatchelder) [reply src/AirshipBindings.iOS.Core/ApiDefinitions.cs]: FYI - I checked the iOS SDK source, and it does indeed contain a unicode "-" which Xamboni converts into that "?". We should probably have a ticket to clean unicode out of the SDK.

Comment (marc-scig) [reply src/AirshipBindings.iOS.Core/ApiDefinitions.cs]: There's already a ticket for that.

**Pull request merged**